src/o365/spo/commands/web/web-add.ts

```diff
@@ -0,0 +1,313 @@
+        .then((perm:BasePermissions) : any => {
+            /// Detects if the site in question has no script enabled or not. 
+            /// Detection is done by verifying if the AddAndCustomizePages permission is missing.
+            /// 
+            /// See https://support.office.com/en-us/article/Turn-scripting-capabilities-on-or-off-1f2c515f-5d7e-448a-9fd7-835da935584f
+            /// for the effects of NoScript
+            /// 
+            if(perm.has(PermissionKind.AddAndCustomizePages)) {
+              return this.setInheritNavigation(siteAccessToken, cmd, args, cb, this.debug).then((res : any) => {
+                cb();
+              },(err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+            }
+            else {
+              cmd.log("No script is enabled. Skipping the InheritParentNavigation settings.")
+              cb();
+            }
+          }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+      }
+      else
+       {
+         cb();
+       }
+    }, (err: any): void => this.handleRejectedODataJsonPromise(err, cmd, cb));
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-t, --title <title>',
+        description: 'Subsite title'
+      },
+      {
+        option: '-d, --description [description]',
+        description: 'Subsite description, optional'
+      },
+      {
+        option: '-u, --webUrl [webUrl]',
+        description: 'Subsite relative url'
+      },
+      {
+        option: '-w, --webTemplate [webTemplate]',
+        description: 'Subsite template, eg. STS#0 (Classic team site)'
+      },
+      {
+        option: '-p, --parentWebUrl [parentWebUrl]',
+        description: 'URL of the parent site under which to create the subsite'
+      },
+      {
+        option: '-l, --locale [locale]',
+        description: 'Subsite locale LCID, eg. 1033 for en-US'
+      },
+      {
+        option: '--breakInheritance [breakInheritance]',
+        description: 'Set to not inherit permissions from the parent site, optional'
+      },
+      {
+        option: '--inheritNavigation [inheritNavigation]',
+        description: 'Set to inherit the navigation from the parent site, optional'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(): CommandValidate {
+    return (args: CommandArgs): boolean | string => {
+     
+      if (!args.options.title) {
+        return 'Required option title missing';
+      }
+
+      if (!args.options.webUrl) {
+        return 'Required option webUrl missing';
+      }
+
+      if (!args.options.webTemplate) {
+        return 'Required option webTemplate missing';
+      }
+
+      if (!args.options.parentWebUrl) {
+        return 'Required option parentWebUrl missing';
+      }
+
+      if (!args.options.locale) {
```

Comment:
Do we need to require locale or could we say that `1033` is the default if none specified? The original spec listed locale as an optional parameter, but wasn't explicit about the default value.